src/api/base/index.ts

```diff
@@ -74,19 +101,33 @@ export default class BaseApi {
 
     const chainId = networkId || (await this.context.chainId)
 
-    let response
-    if (env === undefined) {
+    // `env` is not set, meaning we should try both envs
+    if (!env) {
+      let prodResponse
+      let barnResponse
       try {
-        response = await fetchFn(url, method, `${prodUri[chainId]}/${this.API_VERSION}`, data, requestOptions)
+        // Prod goes first
+        prodResponse = await fetchFn(url, method, `${prodUri[chainId]}/${this.API_VERSION}`, data, requestOptions)
       } catch (error) {
-        response = await fetchFn(url, method, `${barnUri[chainId]}/${this.API_VERSION}`, data, requestOptions)
+        barnResponse = await fetchFn(url, method, `${barnUri[chainId]}/${this.API_VERSION}`, data, requestOptions)
+      }
+      // If `barnResponse` is set it means `prod` fetching threw and `barn` succeeded
+      // Return that, then
+      if (barnResponse) {
+        return barnResponse
+      }
+      // Prod barnResponse can fail without throwing
+      if (prodResponse?.ok) {
```

Comment:
This is the main fix.
It was counting only on the response failing, not returning a 400, so `barn` was never attempted